impeller/scene/scene_context.h

```diff
@@ -46,52 +43,82 @@ class SceneContext {
 
   bool IsValid() const;
 
+  std::shared_ptr<Pipeline<PipelineDescriptor>> GetPipeline(
+      PipelineKey key,
+      SceneContextOptions opts) const;
+
   std::shared_ptr<Context> GetContext() const;
 
   std::shared_ptr<Texture> GetPlaceholderTexture() const;
 
-  std::shared_ptr<Pipeline<PipelineDescriptor>> GetUnlitPipeline(
-      SceneContextOptions opts) const {
-    return GetPipeline(unlit_pipeline_, opts);
-  }
-
  private:
   template <class T>
   using Variants = std::unordered_map<SceneContextOptions,
                                       std::unique_ptr<T>,
                                       SceneContextOptions::Hash,
                                       SceneContextOptions::Equal>;
-  template <class TypedPipeline>
-  std::shared_ptr<Pipeline<PipelineDescriptor>> GetPipeline(
-      Variants<TypedPipeline>& container,
-      SceneContextOptions opts) const {
-    if (!IsValid()) {
-      return nullptr;
-    }
 
-    if (auto found = container.find(opts); found != container.end()) {
-      return found->second->WaitAndGet();
+  class PipelineCollection {
```

Comment:
PipelineLibrary maybe for consistency with impeller/entity.

Reply:
I renamed this to `PipelineVariants` to more closely resemble the Entities terminology since it just encapsulates the variants for one set of stages. I'm trying to avoid overloading `PipelineLibrary` (perhaps to my detriment).